### ISSUE TITLE
Server Timeout Error

### ISSUE THREAD

**Issue by sjbonner** [BirdStudiesCanada/naturecounts#4]
I've been testing the package for accessing data from project feederwatch and keep encounting server timeout errors. I've tried on multiple days from multiple computers, but I have yet to download a data set. The following shows the best behaviour I've encountered -- usually the error occurs at the call to nc_count().

> ## Load packages
> library(naturecounts)
> 
> ## Query nature counts for available collections
> nc_count(username = "sbonner")
Please enter password for NatureCounts user 'sbonner' 
🔑 XXXXXXXXXXXX
 OK
   access      collection nrecords
1     yes        ABATLAS1   123364
2     yes        ABATLAS2   201398
3     yes      ABBIRDRECS   357264
4     yes             BBS  5735895
5     yes       BBS50-CAN  3219534
6     yes   BBS50-US-EAST  8067776
7     yes   BBS50-US-WEST  8701116
8     yes   BCATLAS1BE_DO    41412
9     yes  BCATLAS1BE_RAW   344433
10    yes BCATLAS1BE_SUMM   165485
11    yes      BCATLAS1PC   260647
12    yes            BCMA    11291
13    yes           BGRMM      476
14    yes    CMMN-DET-HBO   274839
15    yes   CMMN-DET-VLBO    56287
16    yes      EBUTTERFLY    26192
17    yes            IMMP       43
18    yes         IMMP_A2       24
19    yes         IMMP_BW       19
20    yes            IMQC      761
21    yes   MBATLAS1BE_DO    93876
22    yes  MBATLAS1BE_RAW   326429
23    yes MBATLAS1BE_SUMM   154856
24    yes      MBATLAS1PC   306094
25    yes     MBBA1BE_RAW   142920
26    yes    MBBA1BE_SUMM    93053
27    yes     MBBA2BE_RAW   263809
28    yes    MBBA2BE_SUMM   114616
29    yes         MBBA2PC   127447
30    yes            MEXU     3251
31    yes            MLMP      802
32    yes              MM     4343
33    yes    MONARCHWATCH   145665
34    yes     OBBA1BE_RAW   407287
35    yes    OBBA1BE_SUMM   205816
36    yes     OBBA2BE_RAW   541743
37    yes    OBBA2BE_SUMM   256702
38    yes         OBBA2PC   592425
39    yes            OBFS   324560
40    yes             PFW  3841194
41    yes            PMMM        5
42    yes     RCBIOTABASE    12614
43    yes            WMMM     8981
44    yes            WPWI     3012
> 

> ## Retrieve some data
## Retrieve some data
> test <- nc_data_dl(username = "sbonner",
test <- nc_data_dl(username = "sbonner",
+                    collections = "PFW",
                   collections = "PFW",
+                    years = c(2017,NA),
                   years = c(2017,NA),
+                    info = "Test data for Bayesian Data Analysis class at Western")
                   info = "Test data for Bayesian Data Analysis class at Western")
Using filters: collections (PFW); fields_set (BMDE2.00-min); start_year (2017)
Collecting available records...
  access collection nrecords
1    yes        PFW   595057
Total records: 595,057

Downloading records for each collection:
  PFW
    Records 1 to 5000 / 595057
The server did not respond within 60s. Trying again...
Error: The server has not respond within the 'timeout' specified.
Either try again later or increase the 'timeout' period.

**Comment by steffilazerte** [comment]
Hi @sjbonner!
For some large datasets (yours here has half a million records), it might take the server longer than expected to compile the data, and the default timeout of 60 could be inappropriate (we're still figuring out what these defaults should be). I would suggest trying again, but give it a time out of about 5 or 10 min (300-600 seconds). 

```
test <- nc_data_dl(username = "sbonner", 
                   collections = "PFW", 
                   years = c(2017,NA),
                   info = "Test data for Bayesian Data Analysis class at Western", 
                   timeout = 300)
```

Let me know how that works!

**Comment by sjbonner** [comment]
Hi @steffilazerte! 

Thanks for the quick reply. I tried setting timeout = 600 (why not) and this worked. What was confusing me is that the code often failed at the point of my call to nc_count(). This seems like a simply query, so I expected the default timing of 30 seconds to be sufficient. Maybe it's not so simple. I had to set the timeout to a large value for both nc_count() and nc_data_dl().

Thanks!

Simon
